src/core/public/index.ts

```diff
@@ -136,6 +136,8 @@ export {
   ErrorToastOptions,
 } from './notifications';
 
+export { MountPoint, UnmountCallback } from './utils';
+
```

Comment:
That seems to be the first time we have types in core that are used in different services (here both `Notification` and `Overlay`), so I did put them in `utils`. However not sure that's really the correct place. Maybe we want to create a `core/public/common` or `core/public/shared` ? WDYT ?

Reply:
Just to be sure I understand this correctly:

```ts
export interface App extends AppBase {
  mount: (context: AppMountContext, params: AppMountParameters) => AppUnmount | Promise<AppUnmount>;
}
```

would become
```ts
export interface AppMountParameters extends MountParameters {
  context: AppMountContext;
}

export type AppMounter = Mount<AppMountParameters>;

export interface App extends AppBase {
  mount: AppMounter;
}
```

Is that correct ?

Reply:
Also, as these type/interfaces are going to be imported from other `core/public` modules, is it the correct good practice to put them in `core/public/application` and import it from here in other modules (like `core/public/notifications`), or should we put it in a more 'shared' module ?

Reply:
I'm actually of the opinion we should hold off on doing this until we remove context in #49691. As it is right now, I don't think the examples above are actually quite compatible with the existing AppMount interface. No sense in breaking that interface twice, just to share types.

In terms of where to put the shared types now, I think a `src/core/public/types` file makes sense. We have a similar file on the server-side.